src/extension/tags.c

```diff
@@ -199,6 +200,10 @@ void dd_tags_rinit()
         zend_llist_init(&_appsec_json_frags, sizeof(zend_string *),
             _zend_string_release_indirect, 0);
     }
+
+    // Lets populate the span so we can query from this moment on
+    client_ip = NULL;
+    _add_ancillary_tags();
```

Comment:
I'm not sure this is the best approach nor the right module to cache the IP in. The entire caching should be done in ip_extraction instead, with a wrapper function that takes care of passing  `$_SERVER` and caching both the IP and duplicate headers (in a thread local).